Attempt at stream extensions and recording toggle

cs-endo opened the pull request:
Allows a -r flag to enable recording (off by default) and an argument to change where the stream is directed to. Instructions slightly unclear so this is not final

cs-endo committed:
Attempt at stream extensions and recording toggle

Allows a -r flag to enable recording (off by default) and an argument to change where the stream is directed to. Instructions slightly unclear so this is not final

rmil commented on stream-nvenc.sh:
```diff
@@ -1,8 +1,33 @@
+
+# DEFAULTS
+rtmpServer="" # Set to default location
+
```
I'd say split this variable into two; the base URL and then the default stream path?

cs-endo replied:
Can you give an example of what each could be?

rmil replied:
Something like:
```
rtmpServer="rtmp://stream.ystv.co.uk/"
defaultStream="internal/ob2"
```

rmil commented on stream-nvenc.sh:
```diff
@@ -1,8 +1,33 @@
+                exit 0
+                ;;
+        -r)
+                toFile="-vcodec pores -acodec copy "'"'"Recordings/${filename}.mov"'"'""
```
vcodec and acodec are the old ffmpeg ways to change the codec. If you swap it to: `-c:v prores -c:a copy` we'll be in business

rmil commented on stream-nvenc.sh:
```diff
@@ -1,8 +1,33 @@
+done
+
+echo " Server: $rtmpServer"
+echo " Starting event: $filename"
```
Perhaps, also explicitly indicate that it is recording?

rmil reviewed (changes requested):
Seems pretty good! Just need to have a nosey at my comments and I'll try and get a test to see it works

cs-endo committed:
By Suggestions

rmil committed:
Fix typo and split url into pieces

Split rtmpServer into 2 pieces that get built together. Also add a standard for inputs to give
